lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -635,6 +681,92 @@ def get_docker_environment(env, env_files):
     return sorted(env_list)
 
 
+def convert_duration_to_nanosecond(time_str):
+    """
+    Return time duration in nanosecond.
+    """
+    if not isinstance(time_str, str):
+        raise ValueError('Missing unit in duration - %s' % time_str)
+
+    regex = re.compile(
+        r'^(((?P<hours>\d+)h)?'
+        r'((?P<minutes>\d+)m(?!s))?'
+        r'((?P<seconds>\d+)s)?'
+        r'((?P<milliseconds>\d+)ms)?'
+        r'((?P<microseconds>\d+)us)?)$'
+    )
+    parts = regex.match(time_str)
+
+    if not parts:
+        raise ValueError('Invalid time duration - %s' % time_str)
+
+    parts = parts.groupdict()
+    time_params = {}
+    for (name, value) in parts.items():
+        if value:
+            time_params[name] = int(value)
+
+    delta = timedelta(**time_params)
+    time_in_nanoseconds = int(delta.total_seconds() * 1000000000)
```

Comment:
How about using `(delta.microseconds + (delta.seconds + delta.days * 24 * 3600) * 10**6) * 10**3` instead? That will also work with Python 2.6. ([Source](https://docs.python.org/2/library/datetime.html#datetime.timedelta.total_seconds).)

Reply:
Perfect!